train_dolly.py

```diff
@@ -137,18 +135,40 @@
 local_output_dir = os.path.join(local_training_root, checkpoint_dir_name)
 dbfs_output_dir = os.path.join(dbfs_output_root, checkpoint_dir_name)
 
-num_gpus_flag = ""
-num_gpus = dbutils.widgets.get("num_gpus")
-if num_gpus:
-    num_gpus = int(num_gpus)
-    num_gpus_flag = f"--num_gpus={num_gpus}"
+# pick an appropriate config file
+gpu_family = dbutils.widgets.get("gpu_family")
+root_path = os.getcwd()
+deepspeed_config = os.path.join(root_path, f"config/{gpu_family}_config.json")
 
 tensorboard_display_dir = f"{local_output_dir}/runs"
 
 print(f"Local Output Dir: {local_output_dir}")
 print(f"DBFS Output Dir: {dbfs_output_dir}")
 print(f"Tensorboard Display Dir: {tensorboard_display_dir}")
 
+# configure the batch_size
+batch_size = 3
+if gpu_family == "a100":
+    batch_size = 6
+
+# configure CUDA memory management
+if gpu_family == "v100":
+    os.environ["PYTORCH_CUDA_ALLOC_CONF"] = "max_split_size_mb:64"
+
+# configure num_gpus, if specified
+num_gpus_flag = ""
+num_gpus = dbutils.widgets.get("num_gpus")
+if num_gpus:
+    num_gpus = int(num_gpus)
+    num_gpus_flag = f"--num_gpus={num_gpus}"
+
+# configure floating point arithmetic format
+float_format_flag = ""
+if gpu_family == "a100":
+    float_format_flag = "--bf16=True"
```

Comment:
Actually, we could drive this in the deepspeed config files, rather than set it to 'auto' in the config and bother switching so much here. Simplifies this.

Reply:
Agreed. I'll remove these parameters from the trainer and let's just set it in the configs.